harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -383,12 +384,27 @@ def _train_for_step(
 
             self.context._current_batch_idx = batch_idx
             if self.context.is_epoch_start():
+                epoch_idx = self.get_epoch_idx(batch_idx)
                 for callback in self.callbacks.values():
                     with self.prof.record_timing(
                         f"callbacks.{callback.__class__.__name__}.on_training_epoch_start"
                     ):
-                        callback.on_training_epoch_start()
+                        sig = signature(callback.on_training_epoch_start)
```

Comment:
I approve of this hack.  Can you put a comment here that indicates in which determined version the old form was deprecated?  That way, when it is ancient we'll have a clear reminder of just how ancient it is, and it will be more likely we remember to actually remove the deprecated form.

Reply:
I changed the warning message to include the next version.